docs/upgrade-guide.md

```diff
@@ -41,6 +46,34 @@ Upgrading the cluster in this manner keeps the cluster online and able to servic
 - The cluster is less fault tolerant while a node is offline for an upgrade because the cluster requires a quorum of nodes to be online to service write requests.
 - Replicating large amounts of data to a node can have a performance impact on the Leader in the cluster.
 
+### Breaking changes from version 23.10 and earlier
+
+#### External TCP API removed
+
+The deprecated external TCP API has been removed in 24.2.0. This means that any external clients using the TCP API will no longer work with EventStoreDB versions 24.2.0 and onwards.
+
+A number of configuration options have been removed as part of this. EventStoreDB will not start by default if any of the following options are present in the database configuration:
+- `AdvertiseTcpPortToClientAs`
+- `DisableExternalTcpTls`
+- `EnableExternalTcp`
+- `ExtHostAdvertiseAs`
+- `ExtTcpHeartbeatInterval`
+- `ExtTcpHeartbeatTimeout`
+- `ExtTcpPort`
+- `ExtTcpPortAdvertiseAs`
+- `NodeHeartbeatInterval`
+- `NodeHeartbeatTimeout`
+- `NodeTcpPort`
+- `NodeTcpPortAdvertiseAs`
+
+#### Changes to Filtered $all subscription checkpoint intervals
+
+The behaviour for filtered `$all` checkpoints was changed in 24.2.0. Instead of receiving a checkpoint exactly after the checkpoint interval, a checkpoint will be issued at least once per checkpoint interval:
```

Comment:
I think `exactly after the checkpoint internal` would be clearer as `directly after the checkpoint interval` on lines 71, 72, and 73. Unless I'm misunderstanding what this is trying to say.